build.sbt

```diff
@@ -530,6 +532,12 @@ lazy val buildInfoSettings = Def.settings(
   buildInfoObject := "BuildInfo"
 )
 
+lazy val noPublish = Seq(
+  publishArtifact := false,
+  publish := {},
+  publishLocal := {}
+)
```

Comment:
I reused noPublish in scalametaRoot: https://github.com/scalameta/scalameta/pull/753/files#diff-fdc3abdfd754eeb24090dbd90aeec2ceR24